artview/components/display.py

```diff
@@ -0,0 +1,297 @@
+        gBox_layout.addWidget(self.ent_texc, 5, 0, 1, 1)
+        gBox_layout.addWidget(self.ent_texsz, 5, 1, 1, 1)
+        gBox_layout.addWidget(self.ent_texst, 5, 2, 1, 1)
+
+        groupBox.setLayout(gBox_layout)
+        return groupBox
+
+    #########################
+    #   Selection Methods   #
+    #########################
+
+    def chooseText(self, selection):
+        '''Get Display Text.'''
+##        selection = self.dispCombo.currentIndex()
+        self.choice_key = str(self.dispCombo.currentText())
+        self.choice = self.dispChoiceList[selection]
+        self._rebuild_entry()
+
+        self.dispCombo.setCurrentIndex(selection)
+##        print(selection)
+
+    def _displayHelp(self):
+        '''Display help.'''
+        text = (
+            "<b>Using the ImageTextBox window</b><br><br>"
+            "<i>Choose an existing Text box:</i><br>"
+            "  1. Select instance from drop-down list.<br>"
+            "  2. Modify exisint instance by changing values in "
+            "     Text Value and Position. Click Update to save.<br>"
+            "  3. Create a new Text box by choosing Add Text from drop-down."
+            "<br>"
+            "  4. Delete an instance by selecting and click Delete Text.<br>"
+            "  5. Clear all instances by clicking Clear All Text.<br>"
+            "<br>"
+            )
+        core.common.ShowLongText(text.replace("\n", "<br>"), set_html=True)
+
+    def updateChoice(self):
+        '''Update the Display text box and/or parameters.'''
+        self._check_entries()
+        self.choice = self._get_entries()
+
+        # Create the text instance
+        # Add to list if new instance or overwrite if present
+        if self.choice['text'] not in self.display.disp_text.keys():
+            self.choice['instance'] = self.display.ax.text(
+                self.choice['xpos'], self.choice['ypos'],
+                self.choice['text'], style=self.choice['style'],
+                color=self.choice['col'], fontsize=self.choice['size'])
+            self.dispCombo.addItem(self.choice['text'])
+            self.dispChoiceList.append(self.choice)
+            self.dispCombo.setCurrentIndex(self.dispCombo.count() - 1)
+        else:
+            print("Overwriting text")
+            select = self.dispCombo.findText(self.choice['text'])
+            try:
+                self.dispChoiceList[select]['instance'].remove()
+            except:
+                print("No instance found")
+            self.choice['instance'] = self.display.ax.text(
+                self.choice['xpos'], self.choice['ypos'],
+                self.choice['text'], style=self.choice['style'],
+                color=self.choice['col'], fontsize=self.choice['size'])
+            self.dispChoiceList[select] = self.choice
+            self.dispCombo.setCurrentIndex(select)
+        # Write the updated text item to the Display Text instance
+        self.display.disp_text[self.choice['text']] = self.choice
```

Comment:
this behavior will create uncontrollable and unintuitive behavior, for instance it does not allow two entries of the same text (in different positions).